Mesh_3/include/CGAL/Compact_mesh_cell_base_3.h

```diff
@@ -37,6 +37,61 @@
 # include <atomic>
 #endif
 
+namespace internal_tbb
+{
+//classic pointer{
+//normal
+template<typename T>
+void set_weighted_circumcenter(T t, T value)
+{
+  t = value;
+}
+
+//overload for nullptr
+template<typename T>
+void set_weighted_circumcenter(T t, std::nullptr_t)
+{
+  t = nullptr;
+}
+template<typename T>
+bool compare_weighted_circumcenter(T t)
+{
+  return t == nullptr;
+}
+
+template<typename T>
+void delete_circumcenter(T t )
+{
+  delete t;
+}
+//} atomic {
+//normal
+template<typename T>
+void set_weighted_circumcenter(std::atomic<T>& t, T value)
+{
+  t.load() = value;
+}
+
+//nullptr
+template<typename T>
+void set_weighted_circumcenter(std::atomic<T>& t, std::nullptr_t)
+{
+  t = nullptr;
+}
```

Comment:
You should have overloaded on `std::atomic<T*>` and not `std::atomic<T>`.

Reply:
As it is only used for weighted_circumcenter_ it is the same thing, but I see how it might cause problems if we use these helpers in another context later...